b2sdk/transfer/emerge/emerger.py

```diff
@@ -40,6 +43,28 @@ def __init__(self, services):
         self.services = services
         self.emerge_executor = EmergeExecutor(services)
 
+    @staticmethod
+    def _get_updated_file_info_with_large_file_sha1(
+        file_info: dict,
+        write_intents: List[WriteIntent],
+        emerge_plan: EmergePlan,
+        large_file_sha1: Optional[Sha1HexDigest] = None,
+    ) -> dict:
+        if not emerge_plan.is_large_file():
+            # Emerge plan doesn't construct a large file, no point setting the large_file_sha1
+            return file_info
+
+        if not large_file_sha1 and len(write_intents) == 1:
+            # large_file_sha1 was not given explicitly, but there's just one write intent, perhaps it has a hash
+            large_file_sha1 = write_intents[0].get_large_file_sha1()
```

Comment:
`WriteIntent` holds a stream that could have `sha1` available. `AbstractUploadSource` already has `get_content_sha1` method. Maybe it would be enough to just push it higher instead of creating a new one?